src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/AuthorizationRepository.kt

```diff
@@ -141,6 +135,57 @@ class OrderlyAuthorizationRepository : AuthorizationRepository
 
     }
 
+    override fun ensureUserGroupDoesNotHavePermission(userGroup: String, permission: ReifiedPermission)
+    {
+        JooqContext().use {
+            checkPermissionExists(it, permission.name)
+            checkUserGroupExists(it, userGroup)
+
+            //delete the permission if exists
```

Comment:
Just a general point - I'm in favour of only using code comments when something is not self documenting. E.g. below the comment about deleting the abstract permission seems reasonable to me as it might not be obvious what's going on from the code. But I think "delete the permissions if exists" is implied by the function name and the code!

Reply:
Should add, I appreciate this is a debated topic and maybe you are in the other camp, but so far it's the approach Martin and I have taken (something along the lines of the accepted answer here: https://softwareengineering.stackexchange.com/questions/51307/self-documenting-code-vs-commented-code). We can revisit the topic if you think it warrants it!